### PR TITLE
AG-16452: Add browser benchmark comparison infrastructure to CI

### PR DESCRIPTION
**Pull request by alantreadway**
## Summary

Adds browser-based benchmark comparison to the CI pipeline, running alongside existing Jest benchmarks. This addresses the disconnect between Node.js benchmark results and real in-browser performance that led to regressions being missed during release testing.

### New files
- **`tools/benchmark/format-utils.js`** — shared formatting utilities (extracted from `format-slack-message.js`)
- **`tools/benchmark/compare-browser-results.js`** — compares two browser benchmark JSON reports, produces ranked timing tables
- **`tools/benchmark/format-browser-slack-message.js`** — formats comparison results as Slack Block Kit JSON
- **`tools/benchmark/compare-browser-latest.sh`** — CI orchestrator: benchmarks HEAD and base branch (via git worktree), then compares

### Modified files
- **`tools/benchmark/format-slack-message.js`** — refactored to use shared `format-utils.js`
- **`.github/workflows/benchmark.yml`** — Playwright install, browser benchmark steps, split PR comment/Slack reporting, artifacts

### Architecture
- `browser-benchmark.ts` always runs from the HEAD working directory (single Playwright install)
- Base branch built in a git worktree with shared Nx cache
- Strictly sequential dev server lifecycle (start → benchmark → stop per branch)
- PR comments show browser results as primary, Jest in collapsible `<details>`
- Overnight Slack gets separate Jest and browser benchmark messages

## Test plan
- [x] Syntax checks pass for all new/modified files
- [x] Smoke test: comparison script with synthetic data (including division-by-zero edge case)
- [x] Smoke test: Slack formatter produces valid Block Kit JSON
- [ ] Manual `workflow_dispatch` trigger to verify end-to-end CI flow
- [ ] `/benchmarks` PR comment trigger to verify PR reporting
- [ ] Overnight scheduled run to verify Slack `#charts-performance` output

AG-16452

### CODE REVIEW

File: tools/benchmark/compare-browser-latest.sh [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    output="${root}/reports/browser-benchmark.json"
+else
+    output="${root}/reports/browser-benchmark.log"
+fi
+
+if [[ -z "$base_results" || ! -f "$base_results" ]]; then
+    log "Base benchmark results unavailable, writing partial report..."
+    echo "Base benchmark results unavailable for ${base_ref}" > "$output"
+    echo "Only head results available at ${head_results}" >> "$output"
+    cat "$output"
```
[the remark this answers is not in the document]
Comment (alantreadway) [reply tools/benchmark/compare-browser-latest.sh:406]: Fixed

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -46,6 +46,7 @@ jobs:
 
     benchmark:
         runs-on: macos-benchmark
+        timeout-minutes: 120
```
Comment (alantreadway) [inline .github/workflows/benchmark.yml]: Reduce this to a tighter timeframe, with a comment to reduce further once we remove the `jest`-based benchmarks

Comment (alantreadway) [reply .github/workflows/benchmark.yml]: Fixed

**Review by iMoses — APPROVED** [review]
LGTM
